stellargraph/layer/rgcn.py

```diff
@@ -0,0 +1,496 @@
+                    activation=self.activations[ii],
+                    use_bias=self.bias,
+                    final_layer=ii == (n_layers - 1),
+                    **self._regularisers
+                )
+            )
+
+    def _get_regularisers_from_keywords(self, kwargs):
+        regularisers = {}
+        for param_name in [
+            "kernel_initializer",
+            "kernel_regularizer",
+            "kernel_constraint",
+            "bias_initializer",
+            "bias_regularizer",
+            "bias_constraint",
+        ]:
+            param_value = kwargs.pop(param_name, None)
+            if param_value is not None:
+                regularisers[param_name] = param_value
+        self._regularisers = regularisers
+
+    def __call__(self, x):
+        """
+        Apply a stack of RGCN layers to the inputs.
+        The input tensors are expected to be a list of the following:
+        [Node features shape (1, N, F), Output indices (1, O)] +
+        [Adjacency indices for each relationship (1, E, 2) for _ in range(R)]
+        [Adjacency values for each relationshiop (1, E) for _ in range(R)]
+
+
+        where N is the number of nodes, F the number of input features,
+              E is the number of edges, O the number of output nodes,
+              R is the number of relationships in the graph (edge types).
+
+        Args:
+            x (Tensor): input tensors
+
+        Returns:
+            Output tensor
+        """
+        x_in, out_indices, *As = x
+
+        # Currently we require the batch dimension to be one for full-batch methods
+        batch_dim, n_nodes, _ = K.int_shape(x_in)
+        if batch_dim != 1:
+            raise ValueError(
+                "Currently full-batch methods only support a batch dimension of one"
+            )
+
+        N_edge_types = len(self.generator.As)
+
+        # Convert input indices & values to sparse matrices
+        if self.use_sparse:
+
+            As_indices = As[:N_edge_types]
+            As_values = As[N_edge_types:]
+
+            Ainput = [
+                SqueezedSparseConversion(
+                    shape=(n_nodes, n_nodes), dtype=As_values[i].dtype
+                )([As_indices[i], As_values[i]])
+                for i in range(N_edge_types)
+            ]
+
+        # Otherwise, create dense matrices from input tensor
+        else:
+            Ainput = [Lambda(lambda A: K.squeeze(A, 0))(A_) for A_ in As]
+
+        h_layer = x_in
+
+        for layer in self._layers:
+            if isinstance(layer, RelationalGraphConvolution):
+                # For an RGCN layer add the adjacency matrices and output indices
+                # Note that the output indices are only used if `final_layer=True`
+                h_layer = layer([h_layer, out_indices] + Ainput)
+            else:
+                # For other (non-graph) layers only supply the input tensor
+                h_layer = layer(h_layer)
+
+        return h_layer
+
+    def node_model(self):
```

Comment:
We are trying to standardise on replacing `node_model()` with `build()`. The goal is to make `build()` return the correct model, node or link, based on the generator. Using `build()` will require the user to change fewer lines of code to re-use their scripts between tasks. Note that not all models have `build()` implemented yet, but we want to provide this method for all new model classes.

Reply:
Should I use `build() `to call `node_model()` is the generator is node generator or removing the `node_model()` function entirely?

Reply:
I think it might be best to remove node_model all together by just making it a "private" method we can call from build().

Reply:
done!